src/imitation/algorithms/sqil.py

```diff
@@ -0,0 +1,321 @@
+            policy: The policy model to use (SB3).
+            custom_logger: Where to log to; if None (default), creates a new logger.
+            learning_rate: The learning rate, it can be a function
+                of the current progress remaining (from 1 to 0).
+            buffer_size: Size of the replay buffer.
+            learning_starts: How many steps of the model to collect transitions for
+                before learning starts.
+            batch_size: Minibatch size for each gradient update.
+            tau: The soft update coefficient ("Polyak update", between 0 and 1),
+                default 1 for hard update.
+            gamma: The discount factor.
+            train_freq: Update the model every ``train_freq`` steps. Alternatively
+                pass a tuple of frequency and unit
+                like ``(5, "step")`` or ``(2, "episode")``.
+            gradient_steps: How many gradient steps to do after each
+                rollout (see ``train_freq``).
+                Set to ``-1`` means to do as many gradient steps as steps done
+                in the environment during the rollout.
+            replay_buffer_class: Replay buffer class to use
+                (for instance ``HerReplayBuffer``).
+                If ``None``, it will be automatically selected.
+            replay_buffer_kwargs: Keyword arguments to pass
+                to the replay buffer on creation.
+            optimize_memory_usage: Enable a memory efficient variant of the
+                replay buffer at a cost of more complexity.
+            target_update_interval: Update the target network every
+                ``target_update_interval``  environment steps.
+            exploration_fraction: Fraction of entire training period over
+                which the exploration rate is reduced.
+            exploration_initial_eps: Initial value of random action probability.
+            exploration_final_eps: Final value of random action probability.
+            max_grad_norm: The maximum value for the gradient clipping.
+            tensorboard_log: The log location for tensorboard (if None, no logging).
+            policy_kwargs: Additional arguments to be passed to the policy on creation.
+            verbose: Verbosity level: 0 for no output, 1 for info messages
+                (such as device or wrappers used), 2 for debug messages.
+            seed: Seed for the pseudo random generators.
+            device: Device (cpu, cuda, ...) on which the code should be run.
+                Setting it to auto, the code will be run on the GPU if possible.
+            _init_setup_model: Whether or not to build the network
+                at the creation of the instance.
+
+
+        """
+        self.venv = venv
+
+        super().__init__(demonstrations=demonstrations, custom_logger=custom_logger)
+
+        self.orig_train_freq = train_freq
+
+        self.dqn = dqn.DQN(
+            policy=policy,
```

Comment:
Could do `*args, **kwargs`. Although then function would be less self-documenting, so I'm unsure if that's better. More maintainable though if DQN changes underneath us?

Reply:
I was also thinking about sort of a middle ground `dqn_kwargs`, but not sure which options is the best

Reply:
I changed to `dqn_kwargs`, not sure about having the caller construct DQN, it could lead to some subtle bugs with the new structure.

Following the other suggestion, I changed it so that we only overwrite `ReplayBuffer`, which can be passed directly into the `DQN` constructor. But this means that if someone just creates a DQN with default settings, then everything could theoretically work, except it's using a normal replay buffer, and therefore running normal DQN. 

To avoid that, we could explicitly check if it's using the right replay buffer, but at that point it feels a bit overcomplicated, so for now I'm leaning towards creating it inside of the algorithm. If you're certain the other approach would be better, I can change it though.

I would probably consider a bring-your-own-algo approach for a potential future implementation/refactoring which would automatically work with all `OffPolicyAlgorithm`s, but this is probably out of scope for now, and has a few extra nuances and edge cases we need to think about carefully. For example, HER uses its own replay buffer implementation, and dictionary obs spaces use a `DictReplayBuffer`. We'd need to be sure that these edge cases at the very least fail loudly.